### PR TITLE
REFACTOR-#5417: fix FutureWarning: the `mangle_dupe_cols` keyword is deprecated

### LINKED ISSUE #5417: CLN: FutureWarning: the `mangle_dupe_cols` keyword is deprecated for `read_csv`
This warning appears for `hdk` engine.

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5417 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/experimental/core/execution/native/implementations/hdk_on_native/io/io.py [hunk 1]
```diff
@@ -129,8 +74,6 @@ class HdkOnNativeIO(BaseIO, TextFileDispatcher):
         "skipfooter",
         "nrows",
         "skipinitialspace",
-        "squeeze",
-        "mangle_dupe_cols",
```
Comment (anmyachev) [inline modin/experimental/core/execution/native/implementations/hdk_on_native/io/io.py:133]: These parameters are handled at a higher level.

File: modin/experimental/core/execution/native/implementations/hdk_on_native/io/io.py [hunk 2]
```diff
@@ -419,15 +315,15 @@ def _read_csv_check_support(
         str
             Error message that should be raised if user explicitly set `engine="arrow"`.
         """
-        filepath_or_buffer = read_csv_kwargs.get("filepath_or_buffer", None)
```
Comment (anmyachev) [inline modin/experimental/core/execution/native/implementations/hdk_on_native/io/io.py:422]: All parameters of the function are defined up to this point. No need to use `get` function, resulting in duplicate default values.

File: modin/experimental/core/execution/native/implementations/hdk_on_native/io/io.py [hunk 3]
```diff
@@ -394,8 +288,10 @@ def _prepare_pyarrow_usecols(cls, read_csv_kwargs):
 
         return usecols_md
 
+    from modin.pandas import read_csv as general_read_csv
```
Comment (anmyachev) [inline modin/experimental/core/execution/native/implementations/hdk_on_native/io/io.py:291]: The most complete function signature is at the top level, we can reuse it.

**Review by ienkovich — APPROVED** [review]

File: modin/experimental/core/execution/native/implementations/hdk_on_native/io/io.py [hunk 4]
```diff
@@ -227,44 +114,51 @@ def read_csv(
         -----
         Reading performed by using of `pyarrow.read_csv` function.
         """
-        items = locals().copy()
-        mykwargs = {k: items[k] for k in items if k in cls.arg_keys}
-        eng = str(engine).lower().strip()
+        eng = str(kwargs["engine"]).lower().strip()
         try:
             if eng in ["pandas", "c"]:
```
Comment (vnlitvinov) [inline modin/experimental/core/execution/native/implementations/hdk_on_native/io/io.py:119]:
```suggestion
            if eng in ("pandas", "c"):
```
let's not make lists when they're not needed - they're slower and take more memory

File: modin/experimental/pandas/io.py [hunk 5]
```diff
@@ -226,6 +226,9 @@ def parser_func(
         _, _, _, f_locals = inspect.getargvalues(inspect.currentframe())
         if f_locals.get("sep", sep) is False:
             f_locals["sep"] = "\t"
+        # mangle_dupe_cols has no effect starting in pandas 1.5. Exclude it from
+        # kwargs so pandas doesn't spuriously warn people not to use it.
+        f_locals.pop("mangle_dupe_cols", None)
```
[the remark this answers is not in the document]
Comment (anmyachev) [reply modin/experimental/pandas/io.py:231]: By using `inspect.getargvalues(inspect.currentframe())`.

File: modin/experimental/core/execution/native/implementations/hdk_on_native/io/io.py (hunk 3 above)
Comment (anmyachev) [reply modin/experimental/core/execution/native/implementations/hdk_on_native/io/io.py]: Good idea, let's try

**Comment by anmyachev** [comment]
@vnlitvinov ready for review

**Comment by anmyachev** [comment]
@ienkovich if you are satisfied with the code, please approve it again :)

**Review by ienkovich — APPROVED** [review]

**Comment by anmyachev** [comment]
@vnlitvinov is there anything else I need to fix?

**Review by mvashishtha — DISMISSED** [review]

**Review by mvashishtha — APPROVED** [review]

**Comment by anmyachev** [comment]
@mvashishtha could we merge it?

**Comment by mvashishtha** [comment]
@anmyachev I'm missing HDK approval

**Pull request merged**